### PR TITLE
Visualizer-support

### PR DESCRIPTION
**Pull request by edan-bainglass**
Various changes and bug fixes required to support epfl-theos/aiidalab-aurora#42.

### CODE REVIEW

File: aiida_aurora/utils/cycling_analysis.py [hunk 1]
```diff
@@ -357,8 +356,29 @@ def add_analysis(data: dict) -> str:
...
-    # TODO replace str(data) with something insightful, and clean!
-    return str(data) if data else "ERROR! Failed to find or parse output"
+
+    if data:
+
+        COLUMNS = {
+            "time": "Time (s)",
+            "I": "I (A)",
+            "Ewe": "Ewe (V)",
+        }
```
Comment (lorisercole) [inline aiida_aurora/utils/cycling_analysis.py:369]: Just wondering (I do not remember) whether the units of `data` are always these ones (s, A, V).

File: aiida_aurora/utils/parsers.py [hunk 2]
```diff
@@ -71,6 +65,7 @@ def get_data_from_results(array_node):
         'I': I,
         'cn': len(Qd),
         'time-cycles': t[idx[2::2]],
-        'Qd': np.array(Qd),
-        'Qc': np.array(Qc),
+        'Q': cumtrapz(I, t, axis=0, initial=0) / 3.6,
+        'Qd': np.array(Qd) / 3.6,
+        'Qc': np.array(Qc) / 3.6,
```
Comment (lorisercole) [inline aiida_aurora/utils/parsers.py:70]: Here have you changed the units of `Q`, `Qd`, and `Qc`?

File: pyproject.toml [hunk 3]
```diff
@@ -25,13 +25,13 @@ classifiers = [
...
 keywords = ["aiida", "plugin", "aurora"]
-requires-python = ">=3.8"
+requires-python = ">=3.9"
 dependencies = [
     "aiida-core>=2.0,<3",
     "aiida-ssh2win>=0.2",
     "dgbowl-schemas>=111,<=113",
     "pydantic<2.0",
-    "matplotlib",
+    "scipy~=1.11.2",
```
Comment (lorisercole) [inline pyproject.toml:34]: Not really important, but this version requirement for scipy seems quite strict.

**Review by lorisercole — APPROVED** [review]
LGTM
I have just added a couple of comments. Of course I have no way to test it, so let Francisco or Peter review as well, if you deem it necessary.

File: aiida_aurora/utils/cycling_analysis.py (hunk 1 above)
Comment (edan-bainglass) [reply aiida_aurora/utils/cycling_analysis.py:369]: When using the AiiDA-Aurora workflow, as far as I know, yes. When running via EC-lab, for example, current may be in mA, in which case the new raw-results importing mechanics (see epfl-theos/aiidalab-aurora#60) would need to start from pre-processed JSON files that converted current to A. This is understood already from experience and will be added to the documentation.

File: aiida_aurora/utils/parsers.py (hunk 2 above)
Comment (edan-bainglass) [reply aiida_aurora/utils/parsers.py:70]: Throughout the app, we treat these charges as capacities. When we do, we generally want them in mAh (ignoring weight normalization for a moment).

File: pyproject.toml (hunk 3 above)
Comment (edan-bainglass) [reply pyproject.toml:34]: I suppose so 😅 I really only need `cumtrapz`. Let me find the minimum `scipy` version that includes a reasonable implementation of the function.

Comment (edan-bainglass) [reply pyproject.toml:34]: Hmm, the function became available in 0.12. However, I've had difficulties installing versions less than 1. How about `scipy>=1.0,<2`? This should ensure support flexibly, while avoiding future breaking changes.

Comment (lorisercole) [reply pyproject.toml:34]: Yes, something like that would be good. Just make sure that it works with scipy 1.0

Comment (edan-bainglass) [reply pyproject.toml:34]: So I tried installing 1.0. Same issues as with <1. The earliest I am able to install is 1.5.4, which works for the app. Pinning `scipy>=1.5.4,<2`

**Pull request merged**